net/rpmsg/rpmsg_sockif.c

```diff
@@ -991,7 +1021,7 @@ static ssize_t rpmsg_socket_send_single(FAR struct socket *psock,
   if (total > ipcsize)
     {
       total = ipcsize;
-      len   = ipcsize - sizeof(*msg) + sizeof(uint32_t);
+      len = ipcsize - sizeof(*msg) - sizeof(uint32_t);
```

Comment:
This seems to be a bugfix that is not strictly related to the new feature being added in this PR.
If so, please add it to a dedicated commit.

Reply:
The original code looks good, I drop this change directly.